parquet-avro/src/main/java/org/apache/parquet/avro/AvroReadSupport.java

```diff
@@ -129,10 +129,10 @@ public RecordMaterializer<T> prepareForRead(
       avroSchema = new Schema.Parser().parse(metadata.get(AVRO_READ_SCHEMA_METADATA_KEY));
     } else if (keyValueMetaData.get(AVRO_SCHEMA_METADATA_KEY) != null) {
       // use the Avro schema from the file metadata if present
-      avroSchema = new Schema.Parser().parse(keyValueMetaData.get(AVRO_SCHEMA_METADATA_KEY));
+      avroSchema = new Schema.Parser().setValidateDefaults(false).parse(keyValueMetaData.get(AVRO_SCHEMA_METADATA_KEY));
```

Comment:
Does this issue exist with both `AVRO_SCHEMA_METADATA_KEY` and `OLD_AVRO_SCHEMA_METADATA_KEY`?

Reply:
Blindly disabling validation at all times may introduce new issues silently. Is it safer to add a flag to work around this? In the default case, we should always enable validation.

Reply:
cc Avro experts @Fokko @gszadovszky @clairemcginty

Reply:
This does seem potentially scary - maybe it would be safest to make this a Configuration option that the user can opt into?

Reply:
+1 making it a config option